app/helpers/fhir_helper.rb

```diff
@@ -242,22 +242,22 @@ def close_contact_from_fhir(related_person)
   end
 
   def history_as_fhir(history)
+    history_creator_reference = FHIR::Reference.new
+    history_creator_reference.identifier = history.created_by
     FHIR::Provenance.new(
       meta: FHIR::Meta.new(lastUpdated: history.updated_at.strftime('%FT%T%:z')),
       id: history.id,
-      patient_id: history.patient_id,
-      occurrenceDateTime: history.created_at,
-      recorded: history.updated_at,
-      reason: [
-        {
-          text: history.comment
-        }
-      ],
+      target: FHIR::Reference.new(reference: "/fhir/r4/Patient/#{history.patient_id}")
+      recorded: history.created_at,
```

Comment:
This needs to be formatted similar to `updated_at`, so using the `strftime('%FT%T%:z')` function.